Modules/_zstd/decompressor.c

```diff
@@ -425,56 +362,29 @@ decompressor_reset_session(ZstdDecompressor *self,
     self->in_begin = 0;
     self->in_end = 0;
 
-    if (type == TYPE_DECOMPRESSOR) {
-        Py_CLEAR(self->unused_data);
-    }
+    Py_CLEAR(self->unused_data);
 
     /* Reset variables in one operation */
     self->needs_input = 1;
-    self->at_frame_edge = 1;
     self->eof = 0;
-    self->_unused_char_for_align = 0;
 
     /* Resetting session never fail */
     ZSTD_DCtx_reset(self->dctx, ZSTD_reset_session_only);
 }
 
 static PyObject *
-stream_decompress(ZstdDecompressor *self, Py_buffer *data, Py_ssize_t max_length,
-                  decompress_type type)
+stream_decompress(ZstdDecompressor *self, Py_buffer *data, Py_ssize_t max_length)
 {
     Py_ssize_t initial_buffer_size = -1;
```

Comment:
`initial_buffer_size` is now a constant `-1`, so you can also simplify the implementation of `decompress_impl`.